lib/cosmos/system/system.rb

```diff
@@ -776,9 +778,10 @@ def save_configuration
         configuration = find_configuration(@config.name)
         configuration = File.join(@paths['SAVED_CONFIG'], File.build_timestamped_filename([@config.name], '.zip')) unless configuration
         unless File.exist?(configuration)
+          configuration_tmp = File.join(@paths['SAVED_CONFIG'], File.build_timestamped_filename(['tmp_' + @config.name], '.zip.tmp'))
```

Comment:
Was there a race condition creating the zip file and someone trying to use it? Is that what the tmp file is for?

Reply:
I was seeing weird failures and realized that changing the system configuration should be way more mutexed.   In particular, yes this temp file prevents a partially written stored config zip file from being used.